integration/test_collection.py

```diff
@@ -790,3 +798,212 @@ def test_empty_search_returns_everything(client: weaviate.Client):
+        CollectionConfig(
+            name="TestFilterNested",
+            vectorizer=Vectorizer.NONE,
+            properties=[Property(name="num", data_type=DataType.NUMBER)],
+            inverted_index_config=InvertedIndexConfigCreate(index_null_state=True),
+        )
+    )
+
+    uuids = [
+        collection.data.insert({"num": 1.0}),
+        collection.data.insert({"num": 2.0}),
+        collection.data.insert({"num": 3.0}),
+        collection.data.insert({"num": None}),
+    ]
+
+    objects = collection.query.get_flat(filters=weaviate_filter)
+    assert len(objects) == len(results)
+
+    uuids = [uuids[result] for result in results]
+    for obj in objects:
+        uuids.remove(obj.metadata.uuid)
+    assert len(uuids) == 0
+
+
+@pytest.mark.skip(
+    "We need to think about how to best do date insertions and returns. The string based comparisons in"
+    "this test sometimes fail due to extra 0s. But filtering works."
+)
+def test_filters_date(client: weaviate.Client):
+    client.collection.delete("TestFilterNested")
+    collection = client.collection.create(
+        CollectionConfig(
+            name="TestFilterDate",
+            vectorizer=Vectorizer.NONE,
+            properties=[Property(name="date", data_type=DataType.DATE)],
+        )
+    )
+
+    now = datetime.datetime.now()
+    collection.data.insert({"date": now}),
+    collection.data.insert({"date": now + datetime.timedelta(hours=1)}),
+
+    objects = collection.query.get_flat(
+        filters=FilterValue(
+            path="date",
+            value=now + datetime.timedelta(minutes=1),
+            operator=FilterOperator.GREATER_THAN,
+        )
+    )
+    assert len(objects) == 1
+    assert objects[0].data["date"] == (now + datetime.timedelta(hours=1)).isoformat("T") + "Z"
+
+
+def test_ref_filters(client: weaviate.Client):
+    client.collection.delete("TestFilterRef")
+    client.collection.delete("TestFilterRef2")
+    to_collection = client.collection.create(
+        CollectionConfig(
+            name="TestFilterRef2",
+            vectorizer=Vectorizer.NONE,
+            properties=[Property(name="int", data_type=DataType.INT)],
+        )
+    )
+    uuid_to = to_collection.data.insert(data={"int": 0})
+    uuid_to2 = to_collection.data.insert(data={"int": 5})
+    from_collection = client.collection.create(
+        CollectionConfig(
+            name="TestFilterRef",
+            properties=[
+                ReferenceProperty(name="ref", reference_class_name="TestFilterRef2"),
+                Property(name="name", data_type=DataType.TEXT),
+            ],
+            vectorizer=Vectorizer.NONE,
+        )
+    )
+
+    from_collection.data.insert({"ref": RefToObject(uuid_to), "name": "first"})
+    from_collection.data.insert({"ref": RefToObject(uuid_to2), "name": "second"})
+
+    objects = from_collection.query.get_flat(
+        filters=FilterValue(
+            path=["ref", "TestFilterRef2", "int"],
```

Comment:
Would it be helpful to introduce a class for this option to help users understand the data that they're expected to provide?